File: lib-database/src/main/java/com/ustadmobile/core/db/dao/VerbDao.java
package com.ustadmobile.core.db.dao;

import com.ustadmobile.lib.database.annotation.UmDao;
import com.ustadmobile.lib.database.annotation.UmQuery;
import com.ustadmobile.lib.database.annotation.UmRepository;
import com.ustadmobile.lib.db.entities.VerbEntity;
import com.ustadmobile.lib.db.sync.dao.SyncableDao;

@UmDao
@UmRepository
public abstract class VerbDao implements SyncableDao<VerbEntity, VerbDao> {


    @UmQuery("SELECT * FROM VerbEntity WHERE urlId = :urlId")
    public abstract VerbEntity findByUrl(String urlId);
}

File: lib-database/src/commonMain/kotlin/com/ustadmobile/core/db/dao/xapi/VerbDao.kt
package com.ustadmobile.core.db.dao.xapi

import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.ustadmobile.door.annotation.DoorDao
import com.ustadmobile.door.annotation.Repository
import com.ustadmobile.lib.db.entities.xapi.VerbEntity

@DoorDao
@Repository
expect abstract class VerbDao  {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    abstract suspend fun insertOrIgnoreAsync(entities: List<VerbEntity>)

    @Query("""
        SELECT VerbEntity.*
          FROM VerbEntity
         WHERE VerbEntity.verbUid = :uid 
    """)
    abstract suspend fun findByUid(uid: Long): VerbEntity?

}